pkg/liquidity-mining/contracts/gauges/StakelessGaugeController.sol

```diff
@@ -0,0 +1,223 @@
+        }
+        emit IStakelessGaugeController.GaugesAdded(gaugeType, gauges);
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#removeGauges.
+     */
+    function removeGauges(IGaugeAdder.GaugeType gaugeType, IStakelessGauge[] calldata gauges)
+        external
+        override
+        gaugeTypeChecked(gaugeType)
+    {
+        EnumerableSet.AddressSet storage gaugesForType = _gauges[gaugeType];
+
+        for (uint256 i = 0; i < gauges.length; i++) {
+            IStakelessGauge gauge = gauges[i];
+            require(gauge.is_killed(), "Gauge was not killed");
+            require(gaugesForType.remove(address(gauge)), "Gauge not present");
+        }
+        emit IStakelessGaugeController.GaugesRemoved(gaugeType, gauges);
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#hasGauge.
+     */
+    function hasGauge(IGaugeAdder.GaugeType gaugeType, IStakelessGauge gauge)
+        external
+        view
+        override
+        gaugeTypeChecked(gaugeType)
+        returns (bool)
+    {
+        return _gauges[gaugeType].contains(address(gauge));
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#getTotalGauges.
+     */
+    function getTotalGauges(IGaugeAdder.GaugeType gaugeType)
+        external
+        view
+        override
+        gaugeTypeChecked(gaugeType)
+        returns (uint256)
+    {
+        return _gauges[gaugeType].length();
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#getGaugeAt.
+     */
+    function getGaugeAt(IGaugeAdder.GaugeType gaugeType, uint256 index)
+        external
+        view
+        override
+        gaugeTypeChecked(gaugeType)
+        returns (address)
+    {
+        return _gauges[gaugeType].at(index);
+    }
+
+    /**
+     * @dev See IStakelessGaugeController#checkpointGaugesAboveRelativeWeight.
+     * Unspent ETH is sent back to sender.
+     */
+    function checkpointGaugesAboveRelativeWeight(uint256 minRelativeWeight) external payable override nonReentrant {
+        (uint256 singleArbGaugeBridgeETH, uint256 totalArbBridgeETH) = _getArbitrumBridgeCosts();
+
+        // solhint-disable-next-line not-rely-on-time
+        uint256 currentPeriod = _roundDownTimestamp(block.timestamp);
+
+        _checkpointGauges(IGaugeAdder.GaugeType.Polygon, minRelativeWeight, currentPeriod, 0);
+        _checkpointGauges(IGaugeAdder.GaugeType.Arbitrum, minRelativeWeight, currentPeriod, singleArbGaugeBridgeETH);
+        _checkpointGauges(IGaugeAdder.GaugeType.Optimism, minRelativeWeight, currentPeriod, 0);
+        _checkpointGauges(IGaugeAdder.GaugeType.Gnosis, minRelativeWeight, currentPeriod, 0);
+        _checkpointGauges(IGaugeAdder.GaugeType.ZKSync, minRelativeWeight, currentPeriod, 0);
+
+        // transfer msg.value - total spent ETH back to sender.
+        // _getArbitrumBridgeCosts ensures the difference is always positive.
+        Address.sendValue(msg.sender, msg.value - totalArbBridgeETH);
```

Comment:
```suggestion
        // Send back any leftover ETH to the caller
        Address.sendValue(msg.sender, address(this).balance);
```

This is both cheaper in terms of gas and less error-prone.

Reply:
Ah, that's a good one. This doesn't need the total eth consumed so far, which is great.